### PR TITLE
Ballistic Bullet Damage & Weapons Overhaul

### PR DESCRIPTION
**Pull request by SouthyOsmanli**
## About The Pull Request

This PR is essentially an overhaul of our ballistic weapon damages in an attempt to balance guns between automatics, semi-automatics and bolt actions.

Many of the bullet damage values come from Fortune/Fortuna with some changes to account for our armor system compared to theirs as well as a few tests regarding ammo types.

The gun templates provided in the PR are from Pebbles on Fortune, giving new coders or those who wish to update the file the proper resources to use templates as not only a reference item but also a base-line of stats.

The new loot-table can be found here: https://docs.google.com/spreadsheets/d/1-GdjC_pcTRLTY7g4oeYYISpphdX_Op88urFXJVDiYs8/edit#gid=0

## Why It's Good For The Game

**Tl;dr** - This PR fixes many of the heavy ‘metas’ for weapons and gives reasons for the usage of semi-automatics and bolt actions over only fully autos. Similarly guns now feel different to use and sport different purposes; such as in the case of 10mm vs .45 APC rounds.

The following points I believe are why this change is beneficial:
- **Decluttering and organization of our code**; removed legacy code, unused or broken guns as well as Citadel/base SS13 guns.
- **Balance of rounds to no longer make certain round types ‘meta’ or feeling broken.** Examples being the .45 cal being outright better than most pistol calibers (10mm, 44 mag, .357, etc) or 5.56 AP rendering guns made for higher pen relatively pointless in comparison. (0.6 pen vs the now lowered 0.33 pen - 0.6 pen was better than AER-9s/2ks AND 7.62 AP)
- **Automatics are now no longer a straight ‘upgrade’ from semi-automatics.** A previous example was how a certain weapons, such as the R-91, felt like a select fire upgrade from the service rifle. Now semi-automatics get damage bonuses; encouraging accurate fire while automatics receive no bonus damage usually and have spread; encouraging sustained fire.
- **Low/mid grade laser weapons now fill a niche.** Before most laser weapons besides the AER-14 could be out shadowed by a 5.56 with AP loaded in them with similar damage and either equal or higher penetration. Lasers now fill a niche of higher pen than normal weapons and, those weapons on par loot-wise with them, sport either roughly equal penetration or higher damage to compensate.
- **Loadouts have been rebalanced mildly** to suit these changes. Example is the R82 being an automatic, thus obtained by the NCR in 3 loadouts (LT, First Sargent, Vet-Ranger). The Legion 'Pelium' is now in 2 Legion loadouts (Vet Deacon, Venetor). The Brotherhood have gotten a lot of balance changes to avoid the faction being too powerful with these changes. Knights lost their AER-9 and gained a 5.56 service rifle variant (nerf), base Paladins have lost their AER-12 for an AER-9 (nerf), Senior Paladin has received the Head Paladin's melee loadout (mild buff) and kept their AER-12 to remain unique and an 'upgrade' from Paladin as they are a team-leader. Finally the Head Paladin has lost their AER-14 in exchange for a gatling laser (better vs mobs, 'worse' vs players ultimately as it has a very low pen and lower damage but tons of laser shots).

## Changelog
:cl:
add: Adds the M14 sprite as a new select-fire 7.62 rifle.
add: Adds a new rifle called the 'Maxson' Carbine; basically a changed service rifle with a modkit that can be made via RnD research.
add: Adds 5mm rounds and mags.
add: Adds AK-112 (5mm) to the games loot spawn.
add: Way more 40x32 sprites; better looking. Examples are the service rifle, LSW, AK112, etc.
balance: Tons of ammo damage changes; from pistols to rifles. Way too many to list; basically nerfed damages in exchange for the guns themselves having bonus damage values applied to them.
balance: Loot tiers re-examined to be concise with the ammunition changes. 
balance: Added a few automatics to loadouts. Examples are the 3 R-82s NCR command staff get. Legion command staff now get the Pilum which is an automatic.  Brotherhood have over-all a soft 'nerf'.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/projectile/bullets/pistol.dm [hunk 1]
```diff
@@ -1,46 +1,81 @@
...
+//////////
+// 9 MM //
+//////////				-Light round, allround
 
-// 9mm
 /obj/item/projectile/bullet/c9mm
 	name = "9mm FMJ bullet"
 	damage = 26
-	armour_penetration = 0.05 //0.15
+	armour_penetration = 0.5
```
Comment (BurpleBineapple) [inline code/modules/projectiles/projectile/bullets/pistol.dm:63]: This seems like a mistake.

File: code/modules/projectiles/projectile/bullets/pistol.dm [hunk 2]
```diff
@@ -52,51 +87,191 @@
...
+	bare_wound_bonus = 36
+
+/obj/item/projectile/bullet/c4570/swc
+	name = ".45-70 SWC bullet"
+	damage = 42
+	armour_penetration = 0.2
+	wound_bonus = 36
+	bare_wound_bonus = 36
+
+/obj/item/projectile/bullet/c4570/explosive
```
Comment (BurpleBineapple) [inline code/modules/projectiles/projectile/bullets/pistol.dm:219]: Meme.

**Review by BurpleBineapple — COMMENTED** [review]
Balance wise I can't say very much, but one major error with making 9mm FMJ the best caliber in the game here.

Also some concern about making lasers nearly hitscan. It's definitely going to tip scales towards laser weaponry because faster projectiles > damage in almost all cases with PVP.

**Comment by Pebblesthe2nd** [comment]
Thats the ammo part of the gun system I made for Fortune, but at a glance the only obvious port of the gun side of the system is the bolt action mag stuff, so it will need a bunch of adjustments beyond the errors Purple spotted. A few things like lasers been changed too and I suppose testing them is needed, but making lasers near hitscan but weak power was one idea I wanted to try, more testing needed as always I suppose, though it looks like the damages has not been adjusted.

Plan was to add special effects to the special ammo types and differentiate more, but its not a bad base to build from.
The values have drifted away from the templates I set though so if they're gonna be used they need updating.

Slug AP for example I was planning to raise a little again, 0.35 was a bit high, 0.2 felt a little low, but it also depends on deflection tresholds etc and I have no clue where DR is at in that regard today.

Assuming its been tested locally (I repathed and renamed about a million things when I made the system so porting does require testing) I do think its a pretty good idea to adopt, but will require a couple follow ups.

**Comment by SouthyOsmanli** [comment]
I've been trying to test it but I'm doing this since Fortuna's guns all have new extra_damage modifiers to balance them alongside this ammo rework I'm assuming. This is very much a work in progress and I'm going to be all the AP values need to be re-adjusted due to our difference in armor systems.

And yeah, I need to really look at slugs since that one did confuse me a bit since it's a little hard to add extra damage on shotgus since it applies to all projectiles - so I'll rework it a bit.

This is a draft though, not a fully-finished product. I'm going to be working on this hopefully more over the weekend.

**Comment by Pebblesthe2nd** [comment]
Yeah I understand Fortuna is implementing some new stuff, this system was designed for something very close to DR baseline though so if you look over the APs one more time it should be good enough I believe. If you do more mods, I suggest simplifying further,  I was considering giving each calibre one special ammo only and work from there, instead of the plethora we got. So for example, the medium pistol cartridge pairing (9mm and 10mm), give 9mm JHP ammo and 10mm AP ammo as their only special ammo. And make JHP have same stats as regular ammo but a wound bonus and being more expensive, perhaps, something along those lines.

Stuff like that! Many of the issues balancing comes from over ambitious amounts of detail and options, having fewer special ammo types but more care put into how they actually work will always be better I am starting to think.

Also, adding + damage to shotties is kinda broken since a +1 to a shotty means +6 damage, at short range with buckshot. To balance shotguns just edit slugs and buck directly is my advice.

**Comment by SouthyOsmanli** [comment]
Feel free to do so! Helps having input on the balancing and suggested changes.

**Comment by AdamElTablawy** [comment]
pretty sure they explicitly asked you not to take any of the contributor's work and it'd be kind of shitty to port without permission

there's no legal ramifications for doing so, since it is open source, but it's a massive dick move

**Comment by SouthyOsmanli** [comment]
> pretty sure they explicitly asked you not to take any of the contributor's work and it'd be kind of shitty to port without permission
> 
> there's no legal ramifications for doing so, since it is open source, but it's a massive dick move

I was told by Pebbles who made the templates it was fine. Baron never asked me anything in DMs. If you tell me who made it I will give credit. 

Never heard a single word.
